app/services/git_based_domain_import_service.rb

```diff
@@ -0,0 +1,20 @@
+class GitBasedDomainImportService
+  def import(git_repo_id, branch_or_tag, tenant_id)
+    git_repo = GitRepository.find_by(:id => git_repo_id)
+
+    ref_type = if git_repo.git_branches.any? { |git_branch| git_branch.name == branch_or_tag }
```

Comment:
I don't know if that's really possible, since every git repo will have at least master, right? I guess, if someone manually crafted the request to not have a branch or tag, then that could happen, but do we really expect anyone to actually do that and complain it doesn't work?